File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/RefInterface.java
package xyz.jienan.xkcd.whatif.interfaces;

import android.webkit.JavascriptInterface;

public class RefInterface {

    private RefCallback callback;

    public RefInterface(RefCallback callback) {
        this.callback = callback;
    }

    @JavascriptInterface
    public void refContent(String refs) {
        if (callback != null) {
            callback.onRefClick(refs);
        }
    }

    public interface RefCallback {
        void onRefClick(String content);
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/RefInterface.kt
package xyz.jienan.xkcd.whatif.interfaces

import android.webkit.JavascriptInterface

class RefInterface(private val callback: RefCallback?) {

    @JavascriptInterface
    fun refContent(refs: String) {
        callback?.onRefClick(refs)
    }

    interface RefCallback {
        fun onRefClick(content: String)
    }
}
